custom-vu/src/main/kotlin/jces1209/vu/page/DcIssueNavigator.kt

```diff
@@ -0,0 +1,52 @@
+package jces1209.vu.page
+
+import jces1209.vu.wait
+import org.openqa.selenium.By
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedConditions.*
+import java.time.Duration
+import java.util.*
+
+
+class DcIssueNavigator(
+    driver: WebDriver
+): IssueNavigator {
+    private val webDriver = driver
+    private val page = FalliblePage.Builder(
+        driver,
+        presenceOfElementLocated(By.className("issue-list"))
+    )
+        .timeout(Duration.ofSeconds(30))
+        .build()
+
+    override fun waitForNavigator() {
+        page.waitForPageToLoad()
+    }
+
+    override fun getIssueElementFromList(): WebElement {
+        val elements = webDriver.wait(
+            presenceOfAllElementsLocatedBy(By.xpath("//*[@class ='issue-list']/*"))
+        )
+        val rndIndex = Random().nextInt(elements.size - 1)
+        val dataKey = elements[rndIndex].getAttribute("data-key")
+        val elementLocator = By.xpath("//*[@data-key = '$dataKey']")
+        webDriver.wait(
+            visibilityOfElementLocated(elementLocator)
+        )
+        webDriver.wait(
+            presenceOfAllElementsLocatedBy(elementLocator)
+        )
+        return webDriver.wait(
+            elementToBeClickable(elementLocator)
+        )
+    }
+
+    override fun selectIssue(element: WebElement) {
+        val dataKey = element.getAttribute("data-key")
+        element.click()
+        webDriver.wait(
```

Comment:
It waits for low number of elements. Did you check by chrome profiler which last elements are being loaded?